evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -172,5 +173,47 @@ def transform(self, X, y=None):
             X_boolean = X.ww[self._boolean_cols.tolist()]
             imputed = self._boolean_imputer.transform(X_boolean)
             X_no_all_null[X_boolean.columns] = imputed
-
+            X_no_all_null = downcast_boolean_nullable_to_double(X_no_all_null)
         return X_no_all_null
+
+
+def downcast_integer_nullable_to_double(X):
+    """Downcasts IntegerNullable types to Double in order to support certain estimators like ARIMA, CatBoost, and LightGBM.
+
+    Args:
+        X (pd.DataFrame): Feature data.
+
+    Returns:
+        X: DataFrame initialized with logical type information where IntegerNullable are cast as Double.
+    """
+    if not isinstance(X, pd.DataFrame):
+        return X
+    X_schema = X.ww.schema
+    original_X_schema = X_schema.get_subset_schema(
+        subset_cols=X_schema._filter_cols(exclude=["IntegerNullable"]),
+    )
+    X_int_nullable_cols = X_schema._filter_cols(include=["IntegerNullable"])
+    new_ltypes_for_int_nullable_cols = {col: "Double" for col in X_int_nullable_cols}
+    X.ww.init(schema=original_X_schema, logical_types=new_ltypes_for_int_nullable_cols)
+    return X
+
+
+def downcast_boolean_nullable_to_double(X):
```

Comment:
shoutout to @ParthivNaresh for these two methods 😄

Reply:
I think we can replace the name to `downcast_boolean_nullable_to_boolean` 😄

Reply:
Also update the description :)

Reply:
@chukarsten the column should have already been imputed (the NaN values shouldn't exist) so these functions are just updated the columns to match the imputed values. It would be from `IntegerNullable` ---> `Integer` or `Double` and `BooleanNullable` to `Boolean`.

Reply:
Duh. Thanks.